plan/relations.go

```diff
@@ -581,6 +593,18 @@ func (p *ProjectRel) CopyWithExpressionRewrite(rewriteFunc RewriteFunc, newInput
 	return &proj, nil
 }
 
+func (p *ProjectRel) ChangeMapping(mapping []int32) error {
+	noutput := int32(len(p.Remap(p.RecordType()).Types))
```

Comment:
If this operation is already mapped, we should be applying the mapping on top of the existing one, not replacing it.

Reply:
I don't see any utility to remap something that's already been mapped.  I could add a check to return an error if someone attempts to remap.